src/analyze/analyze-unit-gdb.c

```
@@ -0,0 +1,120 @@
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Cannot spawn a debugger for a remote service");
+
+        r = unit_name_mangle_with_suffix(argv[1], "as unit", UNIT_NAME_MANGLE_WARN, ".service", &unit);
+        if (r < 0)
+                return log_error_errno(r, "Failed to mangle name '%s': %m", argv[1]);
+
+        r = acquire_bus(&bus, /* use_full_bus= */ NULL);
+        if (r < 0)
+                return bus_log_connect_error(r, arg_transport, arg_runtime_scope);
+
+        _cleanup_free_ char *object = unit_dbus_path_from_name(unit);
+        if (!object)
+                return log_oom();
+
+        r = sd_bus_get_property(
+                        bus,
+                        "org.freedesktop.systemd1",
+                        object,
+                        "org.freedesktop.systemd1.Service",
+                        "MainPID",
+                        &error,
+                        &reply,
+                        "u");
+        if (r < 0)
+                return log_error_errno(r, "Failed to get the main PID of %s: %s", unit, bus_error_message(&error, r));
+
+        pid_t pid;
+        r = sd_bus_message_read(reply, "u", &pid);
+        if (r < 0)
+                return log_error_errno(r, "Failed to read the main PID of %s from reply: %m", unit);
+
+        if (!arg_debugger)
+                arg_debugger = secure_getenv("SYSTEMD_DEBUGGER") ?: "gdb";
+
+        _cleanup_strv_free_ char **debugger_call = NULL;
+        r = strv_extend(&debugger_call, arg_debugger);
+        if (r < 0)
+                return log_oom();
+
+        if (streq(arg_debugger, "gdb")) {
+                r = strv_extendf(&debugger_call, "--pid=" PID_FMT, pid);
+                if (r < 0)
+                        return log_oom();
+        }
+
+        if (streq(arg_debugger, "lldb")) {
+                r = strv_extendf(&debugger_call, "--attach-pid=" PID_FMT, pid);
+                if (r < 0)
+                        return log_oom();
+        }
```

maybe return an error, if it#s neither gdb nor lldb, because we don't know how to handle that.